cnf-certification-test/networking/suite.go

```diff
@@ -67,8 +80,105 @@ var _ = ginkgo.Describe(common.NetworkingTestKey, func() {
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		testMultusNetworkConnectivity(&env, defaultNumPings, netcommons.IPv6)
 	})
+	// Default interface ICMP IPv6 test case
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestListenAndDeclaredIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testListenAndDeclared(&env)
+	})
 })
 
+func parseListening(res string, listeningPorts map[key]string) {
+	var k key
+	lines := strings.Split(res, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if !strings.Contains(line, "LISTEN") {
+			continue
+		}
+		if indexprotocolname > len(fields) || indexport > len(fields) {
+			return
+		}
+		s := strings.Split(fields[indexport], ":")
+		p, _ := strconv.Atoi(s[1])
+		k.port = p
+		k.protocol = strings.ToUpper(fields[indexprotocolname])
+		k.protocol = strings.ReplaceAll(k.protocol, "\"", "")
+		listeningPorts[k] = ""
+	}
+}
+
+func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]string) map[key]string {
+	res := make(map[key]string)
+	if len(listeningPorts) == 0 {
+		return res
+	}
+	for k := range listeningPorts {
+		_, ok := declaredPorts[k]
+		if !ok {
+			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
+			res[k] = listeningPorts[k]
+		}
+	}
+	return res
+}
+
+//nolint:funlen
+func testListenAndDeclared(env *provider.TestEnvironment) {
+	var k key
+	declaredPorts := make(map[key]string)
+	listeningPorts := make(map[key]string)
+	var failedPods string
+	var skippedPods string
+
+	for _, dp := range env.Pods {
+		for i := 0; i < len(env.Containers); i++ {
+			ports := dp.Spec.Containers[0].Ports
+			fmt.Println(ports)
+			if ports == nil {
+				tnf.ClaimFilePrintf("Failed to get declared port for container %d due to %v, skipping pod %s", i, dp.Namespace+"."+dp.Name)
+				skippedPods += env.Containers[i].Namespace + "." + env.Containers[i].NodeName + "\n"
+			}
+			for i := 0; i < len(ports); i++ {
+				k.port = int(ports[i].ContainerPort)
+				k.protocol = string(ports[i].Protocol)
+				declaredPorts[k] = ports[i].Name
+			}
+		}
+
+		oc := clientsholder.GetClientsHolder()
+		output, outerr, err := oc.ExecCommandContainer(clientsholder.Context{Namespace: dp.Namespace,
+			Podname: dp.Name, Containername: dp.Spec.Containers[0].Name}, `ss -tulwnH`)
```

Comment:
You're getting only the listening ports of the first container of the pod (`.Containers[0].Name`). Please move this cal inside the previous loop and call this function for each of the containers of the pod. Then compare both declared and listening ports, as each container in the pod could have different declared/listening ports.

Reply:
This ran on each of the containers of the pod.